### PR TITLE
[TASK] support TYPO3 12

### PR DESCRIPTION
**Pull request by lukasniestroj**
* remove support for TYPO3 version 10.4

github action are copied from sitegeist/fluid-components

### CODE REVIEW

File: Classes/Middleware/StyleguideRouter.php [hunk 1]
```diff
@@ -134,7 +138,7 @@ public function process(
 
             if ($styleguideLanguage) {
                 // Set language in TSFE object
-                $GLOBALS['TSFE']->lang = $styleguideLanguage['identifier'];
```
Comment (lukasniestroj) [inline Classes/Middleware/StyleguideRouter.php:137]: this needs to be reviewed: I am not sure if this is still a leftover of TYPO3 10.4

**Review by ulrichmathes — DISMISSED** [review]
we want to remove compatibility to sitegeist/fluid-components dev-master as its not necessary for TYPO3 v12

File: Classes/Middleware/StyleguideRouter.php [hunk 2]
```diff
@@ -149,13 +151,30 @@ public function process(
...
                 ));
-                $GLOBALS['TYPO3_REQUEST'] = $request;
             }
         }
 
         // Create view
-        $view = $this->createView('fluidStyleguide', 'Styleguide', $actionName);
+        $view = $this->container->get(StandaloneView::class);
+
+        $extbaseAttribute = new ExtbaseRequestParameters();
```
Comment (ulrichmathes) [inline Classes/Middleware/StyleguideRouter.php:160]: The view is not properly initiated in TYPO3 v11.

`Tried resolving a template file for controller action "Standard->index" in format ".html", but none of the paths contained the expected template file (Standard/Index.html). The following paths were checked: /var/www/html/public/typo3conf/ext/fluid_styleguide/Resources/Private/Templates/`

**Review by ulrichmathes — CHANGES_REQUESTED** [review]
Styleguide broken in TYPO3 v11

**Review by ulrichmathes — APPROVED** [review]

**Review by Kayu84 — APPROVED** [review]
works under my tested conditions (11.5 / 12.4)

**Comment by svenpet90** [comment]
Any update on when this will be merged/released? :)

**Pull request merged**